staging/src/k8s.io/component-helpers/node/util/sysctl/namespace.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sysctl
+
+import (
+	"strings"
+)
+
+// Namespace represents a kernel namespace name.
+type Namespace string
+
+const (
+	// refer to https://man7.org/linux/man-pages/man7/ipc_namespaces.7.html
+	// the Linux IPC namespace
+	IPCNamespace = Namespace("IPC")
+
+	// refer to https://man7.org/linux/man-pages/man7/network_namespaces.7.html
+	// the network namespace
+	NetNamespace = Namespace("Net")
+
+	// the zero value if no namespace is known
+	UnknownNamespace = Namespace("")
+)
+
+var namespaces = map[string]Namespace{
+	// kernel semaphore parameters: SEMMSL, SEMMNS, SEMOPM, and SEMMNI.
+	"kernel.sem": IPCNamespace,
+}
+
+var prefixNamespaces = map[string]Namespace{
+	// kernel shared memory limits include shmall, shmmax, shmmni, and shm_rmid_forced.
+	"kernel.shm": IPCNamespace,
+	// kernel messages include msgmni, msgmax and msgmnb.
+	"kernel.msg": IPCNamespace,
+	// mqueue filesystem provides the necessary kernel features to enable the creation
+	// of a user space library that implements the POSIX message queues API.
+	"fs.mqueue.": IPCNamespace,
```

Comment:
Why do some end in dots and some not?  This is basically an API - document it.

Edit: I see what is going on and I hate it.  I'd rather see msgmni, msgmax and msgmnb called out explicitly and require that prefixes be "directories" only.

Reply:
Agree.

I am afraid that we have to keep it as is until we have a better solution.

Reply:
?

Why isn't the better solution to put `msgmni`, `msgmax`, and `msgmnb` into the `nameToNamespace` map ?  Then you can append `.` when comparing the val to the prefix map.

Reply:
That sounds good as there are only 7 items. 
- shmall, shmmax, shmmni, and shm_rmid_forced
- msgmni, msgmax and msgmnb.